includes/service-providers/class-newspack-newsletters-service-provider.php

```diff
@@ -649,13 +649,47 @@ public function update_contact_lists_handling_local( $email, $lists_to_add = [],
 			return true;
 		}
 		if ( static::$support_local_lists ) {
-			$lists_to_add    = $this->update_contact_local_lists( $email, $lists_to_add, 'add' );
-			$lists_to_remove = $this->update_contact_local_lists( $email, $lists_to_remove, 'remove' );
-			if ( is_wp_error( $lists_to_add ) ) {
-				return $lists_to_add;
+			$added_result   = $this->update_contact_local_lists( $email, $lists_to_add, 'add', true );
+			$removed_result = $this->update_contact_local_lists( $email, $lists_to_remove, 'remove', true );
+
+			// If there are errors, log them but allow processing to continue.
+			if ( is_wp_error( $added_result ) ) {
+				do_action(
+					'newspack_log',
+					'newspack_esp_update_contact_lists_error',
+					sprintf(
+						'Error adding contact to lists: %s',
+						implode( ', ', $lists_to_add )
+					),
+					[
+						'type'       => 'error',
+						'data'       => [
+							'provider' => $this->service,
+							'errors'   => $added_result->get_error_messages(),
+						],
+						'user_email' => $email,
+						'file'       => 'newspack_' . $this->service,
+					]
+				);
 			}
-			if ( is_wp_error( $lists_to_remove ) ) {
-				return $lists_to_remove;
+			if ( is_wp_error( $removed_result ) ) {
+				do_action(
+					'newspack_log',
+					'newspack_esp_update_contact_lists_error',
+					sprintf(
+						'Error adding contact to lists: %s',
+						implode( ', ', $lists_to_remove )
+					),
+					[
+						'type'       => 'error',
+						'data'       => [
+							'provider' => $this->service,
+							'errors'   => $removed_result->get_error_messages(),
+						],
+						'user_email' => $email,
+						'file'       => 'newspack_' . $this->service,
+					]
+				);
 			}
 		}
 		return $this->update_contact_lists( $email, $lists_to_add, $lists_to_remove );
```

Comment:
This method expects a modified version of the `$lists_to_add` variable. But you renamed the variable up there and now you're passing the raw value.

Some additional handling is needed there to pass only the remote lists.

Easiest way of doing this is if we use the `Subscription_List` objects instead of the ID, because then you have the `is_local` method and you can call the appropriate method depending on the list type....

You can also get the objects in the `Newspack_Newsletters_Contacts::add_and_remove_lists` method, similar to what we do in the `upsert` method (https://github.com/Automattic/newspack-newsletters/blob/trunk/includes/class-newspack-newsletters-contacts.php#L188). And then this method would receive objects instead of strings

Reply:
I think the renaming of the variable isn't needed, and I don't want to change more than necessary to avoid the errors halting the operation. 16efc54 moves the error logging to the `update_contact_local_lists` method, which is what's causing the issue anyway. This preserves the behavior of modifying the `$lists_to_add` and `$lists_to_remove` arrays while also letting the loop proceed in the event of errors.